paddle/fluid/platform/device_context.cc

```diff
@@ -245,8 +245,15 @@ CUDADeviceContext::CUDADeviceContext(CUDAPlace place)
   eigen_stream_.reset(new EigenCudaStreamDevice());
   eigen_stream_->Reinitialize(&stream_, place);
   eigen_device_.reset(new Eigen::GpuDevice(eigen_stream_.get()));
-  PADDLE_ENFORCE(dynload::cublasCreate(&cublas_handle_));
-  PADDLE_ENFORCE(dynload::cublasSetStream(cublas_handle_, stream_));
+  cublas_handle_.reset(new CublasHandleHolder(stream_, CUBLAS_DEFAULT_MATH));
+
+  if (TensorCoreAvailable()) {
+#if CUDA_VERSION >= 9000
```

Comment:
this seems unnecessary since TensorCoreAvailable already check it.

Reply:
Cannot remove this macro guard, since `CUBLAS_TENSOR_OP_MATH` is not defined in CUDA < 9000.

Reply:
I know, but TensorCoreAvailable() already checks it.

Reply:
No. `CUBLAS_TENSOR_OP_MATH` is not defined in CUDA < 9000. Removing the macro guard would cause compilation error in CUDA < 9000.